src/analysis/seminorm.lean

```diff
@@ -348,6 +387,14 @@ begin
   ...    < r   : by rwa mem_ball_zero at hy,
 end
 
+lemma finset_sup_ball_inter (p : ι → seminorm 𝕜 E) (s : finset ι) (e : E) (r : ℝ) (hr : 0 < r):
+  ball (s.sup p) e r = ⋂ (i ∈ s), ball (p i) e r :=
+begin
+  lift r to nnreal using hr.le,
+  simp_rw [ball, Inter_set_of, finset_sup_apply, nnreal.coe_lt_coe,
+    finset.sup_lt_iff (show ⊥ < r, from hr), ←nnreal.coe_lt_coe, subtype.coe_mk],
+end
+
```

Comment:
I think it's worth also having
```suggestion

lemma ball_finset_sup_eq_finset_inf(p : ι → seminorm 𝕜 E) (s : finset ι) (e : E) (r : ℝ) (hr : 0 < r) :
  ball (s.sup p) e r = s.inf (λ i, ball (p i) e r) :=
begin
  rw finset.inf_eq_infi,
  exact ball_finset_sup_eq_Inter _ _ _ _ hr,
end
```
just because it's nice to have the `finset` spelling on both sides.